### ISSUE TITLE
[2017.7] PY3 test_http_query test is failing on all distros

### ISSUE THREAD

**Issue by rallytime** [saltstack/salt-jenkins#479]
I am fairly certain this is failing due to the boto tests. This started failing after the moto library was updated for Python 3, which was updated to fix #453. This might be related to #478.

**Comment by rallytime** [comment]
More information - the stacktrace looks like this:
```
Traceback (most recent call last):
  File "/testing/salt/utils/templates.py", line 413, in render_jinja_tmpl
    output = template.render(**decoded_context)
  File "/usr/local/lib/python3.5/dist-packages/jinja2/environment.py", line 1008, in render
    return self.environment.handle_exception(exc_info, True)
  File "/usr/local/lib/python3.5/dist-packages/jinja2/environment.py", line 780, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/local/lib/python3.5/dist-packages/jinja2/_compat.py", line 37, in reraise
    raise value.with_traceback(tb)
  File "<template>", line 1, in top-level template code
  File "/testing/salt/utils/http.py", line 314, in query
    method, url, params=params, data=data, **req_kwargs
  File "/usr/lib/python3/dist-packages/requests/sessions.py", line 468, in request
    resp = self.send(prep, **send_kwargs)
  File "/usr/lib/python3/dist-packages/requests/sessions.py", line 576, in send
    r = adapter.send(request, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/moto/packages/responses/responses.py", line 302, in unbound_on_send
    return self._on_request(adapter, request, *a, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/moto/packages/responses/responses.py", line 244, in _on_request
    raise response
requests.exceptions.ConnectionError: Connection refused: GET http://www.google.com/

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/testing/tests/unit/templates/test_jinja.py", line 897, in test_http_query
    dict(opts=self.local_opts, saltenv='test', salt=self.local_salt))
  File "/testing/salt/utils/templates.py", line 460, in render_jinja_tmpl
    trace=tracestr)
salt.exceptions.SaltRenderError: Jinja error: Connection refused: GET http://www.google.com/
Traceback (most recent call last):
  File "/testing/salt/utils/templates.py", line 413, in render_jinja_tmpl
    output = template.render(**decoded_context)
  File "/usr/local/lib/python3.5/dist-packages/jinja2/environment.py", line 1008, in render
    return self.environment.handle_exception(exc_info, True)
  File "/usr/local/lib/python3.5/dist-packages/jinja2/environment.py", line 780, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/local/lib/python3.5/dist-packages/jinja2/_compat.py", line 37, in reraise
    raise value.with_traceback(tb)
  File "<template>", line 1, in top-level template code
  File "/testing/salt/utils/http.py", line 314, in query
    method, url, params=params, data=data, **req_kwargs
  File "/usr/lib/python3/dist-packages/requests/sessions.py", line 468, in request
    resp = self.send(prep, **send_kwargs)
  File "/usr/lib/python3/dist-packages/requests/sessions.py", line 576, in send
    r = adapter.send(request, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/moto/packages/responses/responses.py", line 302, in unbound_on_send
    return self._on_request(adapter, request, *a, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/moto/packages/responses/responses.py", line 244, in _on_request
    raise response
requests.exceptions.ConnectionError: Connection refused: GET http://www.google.com/
```

And this is like the the same bug as reported in https://github.com/spulec/moto/issues/1026

**Comment by rallytime** [comment]
This has been fixed. It was related to the changes made to fix #478.
